android/ui/src/main/java/com/trafi/ui/Badge.kt

```diff
@@ -0,0 +1,258 @@
+            Image(
+                subbadge,
+                modifier = Modifier
+                    .width(constants.subBadgeIconWidth)
+                    .height(constants.subBadgeIconHeight)
+                    .padding(end = 3.dp, bottom = 3.dp)
+                    .constrainAs(createRef()) {
+                        bottom.linkTo(parent.bottom)
+                        top.linkTo(parent.bottom)
+                        start.linkTo(parent.end)
+                        end.linkTo(parent.end)
+                    }
+            )
+        }
+    }
+}
+
+@Composable
+private fun BadgeFiller(
+    badgeType: BadgeType,
+    vector: ImageVector?,
+    text: String?,
+    isHiddenLayoutFiller: Boolean = false,
+    isStackedBadge: Boolean
+) {
+    Row(
+        modifier = Modifier.padding(
+            horizontal = getBadgeHorizontalPadding(badgeType, isStackedBadge),
+            vertical = getBadgeVerticalPadding(badgeType)
+        ),
+        horizontalArrangement = Arrangement.spacedBy(constants.spacer)
+    ) {
+        vector?.let {
+            Image(
+                vector,
+                colorFilter = ColorFilter.tint(AmbientContentColor.current),
+                modifier = Modifier.align(Alignment.CenterVertically)
+                    .width(constants.iconWidth)
+                    .height(if (isHiddenLayoutFiller) 0.dp else constants.iconHeight)
+            )
+        }
+        text?.let {
+            Text(
+                text = text,
+                style = getBadgeTextStyle(badgeType).copy(fontWeight = FontWeight.Bold),
+                modifier = Modifier.align(Alignment.CenterVertically)
+                    .padding(bottom = 1.dp)
+                    .wrapContentHeight(unbounded = true)
+            )
+        }
+    }
+}
+
+@Composable
+private fun getBadgeHeight(type: BadgeType): Dp {
+    return when (type) {
+        SMALL_BADGE -> constants.minHeightSmall
+        MEDIUM_BADGE -> constants.minHeightMedium
+    }
+}
+
+@Composable
+private fun getBadgeRounding(type: BadgeType): Dp {
+    return when (type) {
+        SMALL_BADGE -> constants.cornerRadiusSmall
+        MEDIUM_BADGE -> constants.cornerRadiusMedium
+    }
+}
+
+@Composable
+private fun getBadgeHorizontalPadding(type: BadgeType, isStackedBadge: Boolean): Dp {
+    return when (type) {
+        SMALL_BADGE -> constants.horizontalPaddingSmall + (constants.borderWidth * 2)
+        MEDIUM_BADGE -> constants.horizontalPaddingMedium + (constants.borderWidth * 2)
+    }
+}
+
+@Composable
+private fun getBadgeVerticalPadding(type: BadgeType): Dp {
+    return when (type) {
+        SMALL_BADGE -> 0.dp
+        MEDIUM_BADGE -> constants.verticalPaddingMedium
+    }
+}
+
+@Composable
+private fun getBadgeTextStyle(type: BadgeType): TextStyle {
+    return when (type) {
+        SMALL_BADGE -> constants.textStyleSmall
+        MEDIUM_BADGE -> constants.textStyleMedium
+    }
+}
+
+enum class BadgeType {
+    SMALL_BADGE,
+    MEDIUM_BADGE
```

Comment:
I'd recommend using a slightly more readable naming convention 
```kotlin
enum class BadgeType {
  Small,
  Medium,
}
```
1. avoid repeating "badge"
1. avoid upper snake case for closer consistency with Swift
1. trailing commas 🎉